gen/epilogue.jl

```diff
@@ -34,3 +34,8 @@ waddchstr(win, str) = waddchnstr(win, str, -1)
 
 COLOR_PAIR(n) = NCURSES_BITS(n, 0) & A_COLOR
 PAIR_NUMBER(a) = NCURSES_CAST(Cint, ((NCURSES_CAST(Culong, (a)) & A_COLOR) >> NCURSES_ATTR_SHIFT))
+
+for var in (:curscr, :stdscr, :newscr, :ttytype, :COLORS, :COLOR_PAIRS, :COLS,
+            :ESCDELAY, :LINES, :TABSIZE)
+    @eval $var() = unsafe_load(cglobal(($(QuoteNode(var)), libncurses), Ptr{Cvoid}))
```

Comment:
This introduces Julia functions for what a global variables in C. Is it possible to turn these into Julia constants instead?

That seems only possible if these are compile-time constants. If these are "link-time constants" (e.g. addresses of global variables in C) then this wouldn't work without preventing precompiling the module; that wouldn't be worth it.

Reply:
At least `curscr, stdscr, newscr` are runtime variables which are set or change when calling `initscr, newterm, set_term` etc.

I went for the function approach rather than `const ... = cglobal(...)` as that would then correspond to a `SCREEN **` rather than `SCREEN *`, which is also different from `C`.
Using the function incorrectly in a `ccall` at least makes it obvious what is wrong compared to a wrong pointer.

---

But I think all of that could be addressed with a `LazyCglobal` type (untested)
```
mutable struct LazyCglobal{T<:Ptr}
  p::T
end
Base.cconvert(T::Type{<:Ptr}, l::LazyCglobal) = unsafe_load(l.p)

const stdscr = LazyCglobal{Ptr{Ptr{Cvoid}}}(C_NULL)
function __init__()
   stdscr.p = cglobal((:stdscr, libncurses), Ptr{Cvoid})
end
```